### PR TITLE
[Multi-Actions] Add foreign key to AgentGenerationConfiguration + backfill

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
Step 1 of  #4595 : add fk in AgentGenConfig + backfill

Step 2 will frontfill + migrate + remove null constraint

## Risk
None on this one, a lot on the next

## Deploy Plan
- front init_db
- deploy front

### CODE REVIEW

File: front/lib/models/assistant/agent.ts [hunk 1]
```diff
@@ -228,6 +231,18 @@ AgentConfiguration.belongsTo(AgentGenerationConfiguration, {
   foreignKey: { name: "generationConfigurationId", allowNull: true }, // null = no generation set for this Agent
 });
 
+// NEW -- AgentConfig -> GenerationConfig (1:N)
+AgentConfiguration.hasMany(AgentGenerationConfiguration, {
+  // TODO(@pr) make it non-nullable
+  foreignKey: { name: "agentConfigurationId", allowNull: true },
+  onDelete: "CASCADE",
```
Comment (spolu) [inline front/lib/models/assistant/agent.ts:238]: Make sure to test agent deletion (downgrade and GDPR)

We def want to remove the CASCADE eventually, you may need it during the double dep situation

**Review by spolu — APPROVED** [review]
cc @fontanierh for review as well

File: front/lib/models/assistant/agent.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/models/assistant/agent.ts:238]: When we make it non-nullable you don't really have a choice AFAIK, CASCADE is the default.

**Review by fontanierh — APPROVED** [review]
LGTM !

File: front/lib/models/assistant/agent.ts (hunk 1 above)
Comment (philipperolet) [reply front/lib/models/assistant/agent.ts:218]: Tested, works

**Comment by spolu** [comment]
Where is the migration? Should have come with this PR no?

**Comment by philipperolet** [comment]
It's in #4712 . Hope it's not an issue

**Pull request merged**